forecasting/src/autogluon/forecasting/models/gluonts_model/abstract_gluonts/abstract_gluonts_model.py

```diff
@@ -0,0 +1,287 @@
+import autogluon.core.utils.savers.save_pkl as save_pkl
+import autogluon.core.utils.loaders.load_pkl as load_pkl
+from ...abstract.abstract_model import AbstractModel
+from gluonts.evaluation.backtest import make_evaluation_predictions
+from gluonts.model.forecast import SampleForecast, QuantileForecast
+from pathlib import Path
+import pandas as pd
+import numpy as np
+import copy
+import os
+import time
+from gluonts.evaluation import Evaluator
+from ..abstract_gluonts.model_trial import model_trial
+from tqdm import tqdm
+from autogluon.core.scheduler.fifo import FIFOScheduler
+from autogluon.core.utils.exceptions import TimeLimitExceeded
+from autogluon.core.task.base.base_predictor import BasePredictor
+from gluonts.model.predictor import Predictor
+import logging
+from autogluon.core.constants import AG_ARGS_FIT, BINARY, REGRESSION, REFIT_FULL_SUFFIX, OBJECTIVES_TO_NORMALIZE
+
+logger = logging.getLogger(__name__)
+
+
+class AbstractGluonTSModel(AbstractModel):
+
+    model_file_name = "model.pkl"
+    gluonts_model_path = "gluon_ts"
+
+    def __init__(self, path: str, freq: str, prediction_length: int, name: str, eval_metric: str = None,
+                 hyperparameters=None, model=None, **kwargs):
+        """
+        Create a new model
+        Args:
+            path(str): directory where to store all the model
+            freq(str): frequency
+            name(str): name of subdirectory inside path where model will be saved.
+            eval_metric(str): objective function the model intends to optimize, will use mean_wQuantileLoss by default
+            hyperparameters: various hyperparameters that will be used by model (can be search spaces instead of fixed values).
+        """
+        super().__init__()
+        self.name = name
+        self.path_root = path
+        self.path_suffix = self.name + os.path.sep
+        self.path = self.path_root + self.path_suffix
+
+        if eval_metric is None:
+            eval_metric = "mean_wQuantileLoss"
+        if eval_metric is not None and eval_metric not in ["MASE", "MAPE", "sMAPE", "mean_wQuantileLoss"]:
+            raise ValueError(f"metric {eval_metric} is not available yet.")
+        self.eval_metric = eval_metric
+
+        self.params = {}
+        self.set_default_parameters()
+        self.params["freq"] = freq
+        self.params["prediction_length"] = prediction_length
+
+        self.best_configs = self.params.copy()
+
+        self.nondefault_parameters = []
+        if hyperparameters is not None:
+            self.params.update(hyperparameters)
+            self.nondefault_parameters = list(hyperparameters.keys())[:]
+
+        self.model = model
+        self.val_score = None
+        self.fit_time = None
+        self.predict_time = None
+        self.quantiles = kwargs.get("quantiles", ["0.5"])
```

Comment:
I think here you want to use median? Maybe add an explanation for 0.5 because it's a magic number here